changelog/4727.bugfix.rst

```diff
@@ -0,0 +1,5 @@
+Fix a bug in `~sunpy.map.GenericMap.submap` where only the top right and bottom
```

Comment:
I don't think this is really a bug; we changed the behaviour, and clearly documented that new behaviour in the docstring. But, we did not intend to do that, so now we are reverting to the old behaviour. I think it's a choice of behaviour, rather than a bug per se. There were never any concrete promises in the docstring of what `submap` actually did before I added the new pixel space one.

Reply:
While that might all be true, I can't accept that this is not a bug. That choice of behaviour was incorrect, and by that I mean a unexpected change in behaviour from 1.0.

Reply:
Thinking about this more overnight, I see an argument for not backporting this to 2.0. It is a large behaviour change.

That being said, despite the imminent release of 2.1, we are still on the hook for supporting 2.0 for another 6 months. Are we sure, despite the docstring describing the behaviour we want to maintain the "wrong" behaviour for another 6 months?

I have unlabelled this as backport until we decide what we want to do.

Reply:
I can't make up my mind either way. Worth noting that if we do backport, there is a clear and easy way for users to get back to the old behaviour, which is to do
```python
top_right = smap.world_to_pixel(top_right) * u.pix
bottom_left = smap.world_to_pixel(bottom_left) * u.pix
```
and pass the pixels into `submap`.

Reply:
Just reviewing now, and I think it would be good to put the instructions I listed in the above comment in the changelog, in case anyone wants to use the previous behaviour with coordinates.